### PR TITLE
chore: patches to NLB support 

### PR DESCRIPTION
**Pull request by Lou1415926**
This PR contains a bunch "patch fixes" (quoted because they are not really fixes since NLB isn't released) that surfaced after a few test runs. Each commit fixes an issue, and the commit message should be a summary of what it's fixing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/workload.go [hunk 1]
```diff
@@ -215,6 +215,15 @@ type NetworkLoadBalancerListener struct {
 	SSLPolicy       *string
 	Aliases         []string
 	Stickness       *bool
+	HealthCheck     NLBHealthCheck
+}
+
+type NLBHealthCheck struct {
```
Comment (iamhopaul123) [inline internal/pkg/template/workload.go:221]: comment pls 🙏

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/deploy/cloudformation/stack/transformers.go [hunk 2]
```diff
@@ -321,6 +334,8 @@ func (s *LoadBalancedWebService) convertNetworkLoadBalancer() (networkLoadBalanc
 				TargetPort:      targetPort,
 				SSLPolicy:       nlbConfig.SSLPolicy,
 				Aliases:         aliases,
+				HealthCheck:     hc,
+				Stickness:       nlbConfig.Stickiness,
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/transformers.go]:
```suggestion
				Stickiness:       nlbConfig.Stickiness,
```

File: internal/pkg/template/templates/workloads/partials/cf/nlb.yml [hunk 3]
```diff
@@ -31,38 +31,34 @@ NLBTargetGroup:
...
+    {{- end }}
+    Port: {{ .NLB.Listener.TargetPort }}
+    Protocol: {{- if eq .NLB.Listener.Protocol "TLS"}} TCP {{- else}} {{ .NLB.Listener.Protocol }} {{- end}}
     TargetGroupAttributes:
       - Key: deregistration_delay.timeout_seconds
         Value: {{.DeregistrationDelay}}  # ECS Default is 300; Copilot default is 60.
 {{- if ne .NLB.Listener.Protocol "TLS"}}
-{{- if .NLB.Listener.Stickness }}
-{{/*Sticky sessions are not supported with TLS listeners and TLS target groups.*/}}
+{{- if .NLB.Listener.Stickness }} {{/*Sticky sessions are not supported with TLS listeners and TLS target groups.*/}}
```
Comment (huanjani) [inline internal/pkg/template/templates/workloads/partials/cf/nlb.yml]:
```suggestion
{{- if .NLB.Listener.Stickiness }} {{/*Sticky sessions are not supported with TLS listeners and TLS target groups.*/}}
```

File: internal/pkg/template/templates/workloads/partials/cf/nlb.yml [hunk 4]
```diff
@@ -31,38 +31,34 @@ NLBTargetGroup:
...
       - Key: stickiness.enabled
         Value: {{ .NLB.Listener.Stickness }}
```
Comment (huanjani) [inline internal/pkg/template/templates/workloads/partials/cf/nlb.yml]:
```suggestion
        Value: {{ .NLB.Listener.Stickiness }}
```

File: internal/pkg/template/workload.go [hunk 5]
```diff
@@ -215,6 +215,16 @@ type NetworkLoadBalancerListener struct {
 	SSLPolicy       *string
 	Aliases         []string
 	Stickness       *bool
```
Comment (huanjani) [inline internal/pkg/template/workload.go]:
```suggestion
	Stickiness       *bool
```

**Review by huanjani — APPROVED** [review]
Looks great! Feel free to remove the DNM label after you add the 'i's! 👀

File: internal/pkg/deploy/cloudformation/stack/transformers.go (hunk 2 above)
Comment (Lou1415926) [reply internal/pkg/deploy/cloudformation/stack/transformers.go]: !!yay thank you!!!

**Pull request merged**